master/buildbot/status/web/build.py

```diff
@@ -290,8 +291,22 @@ def content(self, req, cxt):
         cxt['build_url'] = path_to_build(req, b, False)
         cxt['slave_debug_url'] = self.getBuildmaster(req).config.slave_debug_url
         cxt['customBuildUrls'] = b.getCustomUrls()
+        cxt['build_chain_top_build_url'] = None
         codebases_arg = cxt['codebases_arg'] = getCodebasesArg(request=req)
 
+        if b.buildChainID is not None:
+            status = self.getStatus(req)
+
+            build_chain_build_request = yield b.master.db.buildrequests.getBuildRequestById(b.buildChainID)
+            build_chain_builder = status.getBuilder(build_chain_build_request['buildername'])
+
+            if build_chain_builder is not None:
+                build_chain_build_number = yield b.master.db.builds.getBuildNumberForRequest(b.buildChainID)
```

Comment:
there is also this option for generating the URLs, that needs minimum the following parameters: id (in this case buildChainID) , buildername, buildnumber. 

> yield master.status.getURLForBuildRequest

https://github.com/Unity-Technologies/katana/blob/katana/master/buildbot/status/master.py#L182
an example of its ussage is here 
https://github.com/Unity-Technologies/katana/blob/katana/master/buildbot/steps/trigger.py#L220

it will be nice just run  a single query to the db to just fetch the buildername and build number.